system-int-test/test/cmr/system_int_test/search/harvesting_test.clj

```diff
@@ -0,0 +1,77 @@
+  (let [format-key :echo10
+        coll1-echo (data2-core/ingest "PROV1" (data2-collection/collection) {:format :echo10})
+        coll1-concept-id (:concept-id coll1-echo)
+        coll2-echo (data2-core/ingest "PROV1" (data2-collection/collection) {:format :echo10})
+        g1-echo (data2-core/ingest "PROV1" 
+                                   (data2-granule/granule coll1-echo {:granule-ur "g1"
+                                                                      :producer-gran-id "p1"})
+                                   {:format :echo10})
+        g2-echo (data2-core/ingest "PROV1" 
+                                   (data2-granule/granule coll1-echo {:granule-ur "g2"
+                                                                      :producer-gran-id "p2"})
+                                   {:format :echo10})
+        g3-echo (data2-core/ingest "PROV1" 
+                                   (data2-granule/granule coll1-echo {:granule-ur "g3"
+                                                                      :producer-gran-id "p3"})
+                                   {:format :echo10})
+        g4-echo (data2-core/ingest "PROV1" 
+                                   (data2-granule/granule coll1-echo {:granule-ur "g4"
+                                                                      :producer-gran-id "p4"})
+                                   {:format :echo10})
+        g5-echo (data2-core/ingest "PROV1" 
+                                   (data2-granule/granule coll1-echo {:granule-ur "g5"
+                                                                      :producer-gran-id "p5"})
+                                   {:format :echo10})
+        g6-echo (data2-core/ingest "PROV1" 
+                                   (data2-granule/granule coll2-echo {:granule-ur "g6"
+                                                                      :producer-gran-id "p6"})
+                                   {:format :echo10})
+        coll1-grans [g1-echo g2-echo g3-echo g4-echo g5-echo]]
+
+    (index/wait-until-indexed)
+
+    (testing "Harvest by collection-concept-id"
+        (let [params {:collection_concept_id (:concept-id coll1-echo) :scroll true :page-size 2}
+              options {:accept nil
+                       :url-extension "native"}
+              response (search/find-metadata :granule format-key params options)
+              scroll-id (:scroll-id response)]
+          (testing "First search gets expected granules and scroll-id"
+            (is (= (count coll1-grans) (:hits response)))
+            (is (not (nil? scroll-id)))
+            (data2-core/assert-metadata-results-match format-key [g1-echo g2-echo] response))
+    
+          (testing "Second search gets next two granules"
+            (data2-core/assert-metadata-results-match 
+              format-key 
+              [g3-echo g4-echo]
+              (search/find-metadata :granule format-key {:scroll true :scroll-id scroll-id})))
```

Comment:
Probably a good idea. I still need to find the code where we process headers. I think I need to copy the scroll idea from the header into the params, because I don't think the param validations get the headers.